braindecode/models/signal_jepa.py

```diff
@@ -0,0 +1,994 @@
+
+    The difference with a regular :class:`nn.Embedding` is that the embedding
+    vectors are initialized with a positional encodding of the channel locations.
+
+    Parameters
+    ----------
+        locs: list of (list of float or None)
+            List of the n-dimensions locations of the EEG channels.
+        embedding_dim: int
+            Dimensionality of the embedding vectors. Must be a multiple of the number
+            of dimensions of the channel locations.
+    """
+
+    def __init__(self, locs: list[list[float] | None], embedding_dim: int, **kwargs):
+        self.ranges = [
+            (min(col), max(col))
+            for col in zip(
+                *[
+                    row[3:6] if len(row) == 12 else row
+                    for row in locs
+                    if row is not None
+                ]
+            )
+        ]
+        x_min_list, x_max_list = zip(*self.ranges)
+        x_min = min(x_min_list)
+        x_max = max(x_max_list)
+        self.x_max = max(abs(x_min), abs(x_max))
+        self.coord_dim = embedding_dim // len(self.ranges)
+        self.locs = list(locs)
+        print(f"{self.ranges=}")
+        assert embedding_dim % len(self.ranges) == 0
+        super().__init__(len(locs), embedding_dim, **kwargs)
+
+    def reset_parameters(self):
+        for i, loc in enumerate(self.locs):
+            if loc is None:
+                nn.init.zeros_(self.weight[i])
+            else:
+                for j, (x, (x0, x1)) in enumerate(zip(loc, self.ranges)):
+                    with torch.no_grad():
+                        self.weight[
+                            i, j * self.coord_dim : (j + 1) * self.coord_dim
+                        ].copy_(
+                            pos_encode_contineous(
+                                x,
+                                0,
+                                10 * self.x_max,
+                                self.coord_dim,
+                                device=self.weight.device,
+                            ),
+                        )
+
+
+class PosEncoder(nn.Module):
+    """Positional encoder for EEG data.
+
+    Parameters
+    ----------
+        spat_dim: int
+            Number of dimensions to use to encode the spatial position of the patch,
+            i.e. the EEG channel.
+        time_dim: int
+            Number of dimensions to use to encode the temporal position of the patch.
+        ch_names: list[str]
+            List of all the EEG channel names that could be encountered in the data.
+        ch_locs: list of list of float or 2d array
+            List of the n-dimensions locations of the EEG channels.
+        sfreq_features: float
+            The "downsampled" sampling frequency returned by the feature encoder.
+        spat_kwargs: dict
+            Additional keyword arguments to pass to the :class:`nn.Embedding` layer used to
+            embed the channel names.
+    """
+
+    max_seconds: float = 600.0  # 10 minutes
```

Comment:
```suggestion
    max_seconds: float = 600.0  # 10 minutes
```

should be on the init this one, no?